### ISSUE TITLE
Add jna as non optional within the test framework

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#17476]
I'm using the test framework 5.0.0-alpha1 from an external plugin (see `pom.xml` snippet):

```xml
        <dependency>
            <groupId>org.elasticsearch.test</groupId>
            <artifactId>framework</artifactId>
            <version>${elasticsearch.version}</version>
            <scope>test</scope>
        </dependency>
```

When implementing a simple IT like this:

```java
public class BanaPluginTest extends ESIntegTestCase {
    public void testBanoPlugin() {    }
}
```

And launching, I'm getting this error:

```
ERROR   0.00s | BanaPluginTest (suite) <<<
   > Throwable #1: java.lang.NoClassDefFoundError: com/sun/jna/Structure$ByReference
   >    at java.lang.ClassLoader.defineClass1(Native Method)
   >    at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
   >    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
   >    at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
   >    at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
   >    at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
   >    at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
   >    at java.security.AccessController.doPrivileged(Native Method)
   >    at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
   >    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
   >    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
   >    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
   >    at org.elasticsearch.bootstrap.JNANatives.trySetMaxSizeVirtualMemory(JNANatives.java:131)
   >    at org.elasticsearch.bootstrap.Bootstrap.initializeNatives(Bootstrap.java:138)
   >    at org.elasticsearch.bootstrap.BootstrapForTesting.<clinit>(BootstrapForTesting.java:77)
   >    at org.elasticsearch.test.ESTestCase.<clinit>(ESTestCase.java:111)
   >    at java.lang.Class.forName0(Native Method)
   >    at java.lang.Class.forName(Class.java:348)
   >    at com.carrotsearch.randomizedtesting.RandomizedRunner$2.run(RandomizedRunner.java:585)
   > Caused by: java.lang.ClassNotFoundException: com.sun.jna.Structure$ByReference
   >    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
   >    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
   >    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
   >    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
   >    ... 19 more
```

Actually `jna` is defined a an optional library in elasticsearch, which is fine, but the test framework uses it.

To fix that, I have to change my `pom.xml` and add:

```xml
        <dependency>
            <groupId>net.java.dev.jna</groupId>
            <artifactId>jna</artifactId>
            <version>4.1.0</version>
            <scope>test</scope>
        </dependency>
```

I think we should declare `jna` not optional in the test framework so it will come by default when anyone uses it.

**Comment by dadoonet** [comment]
@rjernst WDYT?

**Comment by jimferenczi** [comment]
The direct calls to JNANatives trySetMaxNumberOfThreads and trySetMaxSizeVirtualMemory in bootstrap makes jna non optional on Linux and MAC_OS_X.
I don't think you should add jna for your tests nor declare jna non optional.
@jasontedor couldn't we use the Natives wrapper for those calls ?

**Comment by jasontedor** [comment]
I agree with @jimferenczi and I opened #17492.

**Comment by dadoonet** [comment]
I definitely prefer this idea as well! Thanks guys!

**Comment by jasontedor** [comment]
Closed by #17492 via e85535724ef9a1cfa622fd79a3593d3b714e5570.
